src/main/java/de/tum/in/www1/artemis/web/rest/repository/RepositoryProgrammingExerciseParticipationResource.java

```diff
@@ -316,7 +276,7 @@ public ResponseEntity<Map<String, String>> updateParticipationFiles(@PathVariabl
         // User must have the necessary permissions to update a file.
         // When the buildAndTestAfterDueDate is set, the student can't change the repository content anymore after the due date.
         boolean repositoryIsLocked = programmingExerciseParticipation.isLocked();
-        if (repositoryIsLocked || !participationService.canAccessParticipation(programmingExerciseParticipation)) {
+        if (repositoryIsLocked || !participationService.canAccessParticipation(programmingExerciseParticipation, userRepository.getUserWithGroupsAndAuthorities())) {
```

Comment:
Since there are a lot of changes like this now, would it make sense to proved an overloaded method which then loads the user?

Reply:
That would make sense!